### DISCUSSION TITLE
Multi-Geo

### DISCUSSION THREAD

**Discussion by mark3grahams** [pnp/powershell#3536] [General]
Are there any PnP commands that work like the SPO ones do for Multi-Geo?
Such as Get-SPOGeoMoveCrossCompatibilityStatus and Start-SPOUserAndContentMove

**Comment by AndersRask** [comment]
You can find all cmdlets in this module by typing
`gcm -module pnp.powershell`
AFAIK the only geo specifics are -RestrictedToGeo on Set-PnPSite but I could be wrong

**Comment by mark3grahams** [comment]
I am trying to move users to satellite geo-locations, but I'd like to use an app registration/cert instead of a credential.

Connect-SPOService does not offer that.

**Comment by gibi916** [comment]
Hi,
I'm also interested in using commands like Get-SPOUserAndContentMoveState and Start-SPOUserAndContentMove with PnP.
The reason is that Connect-SPOService isn't really designed for automation scenarios.
Someone have some news about that ?

**Comment by Borgquite** [comment]
+1 for this request. Can't find a way to use these MultiGeo management commands with a PowerShell module that supports app registrations / certificates / managed identities...

https://learn.microsoft.com/en-us/microsoft-365/enterprise/m365-dr-workload-spo?view=o365-worldwide#managing-the-multi-geo-environment

**UPDATE:** Have raised this as a feature request here: https://github.com/pnp/powershell/issues/5103

**Reply by mark3grahams** [reply]
Two years and nothing. I wouldn't get my hopes up.

**Reply by Borgquite** [reply]
Perhaps this is the wrong place to raise it. I've raised an Issue here:

https://github.com/pnp/powershell/issues/5103

**Comment by gautamdsheth** [comment]
Hello @mark3grahams , @gibi916 , @Borgquite , @AndersRask 

Has been a long time, but we finally added 20 multi-geo cmdlets, almost all which are in SPO management shell ! They are now available in the latest nightly builds , please do test and try them out , would love to hear your feedback !

**Reply by Borgquite** [reply]
Thanks so much for all the effort here! As mentioned in #5103 I'm planning to test what I can as soon as current work project slows down :)

**Reply by Borgquite** [reply]
I've done more testing on #5103 and left feedback there - thank you so much for all the hard work, will look forward to using this once it's in prod :)

**Reply by Borgquite** [reply]
All seems to be working to me in the latest nightlies, appreciate all the hard work here @gautamdsheth!
